ex2_services/Services/source/adcs/adcs_service.c

```diff
@@ -1344,6 +1344,22 @@ SAT_returnState adcs_service_app(csp_packet_t *packet) {
         break;
     }
 
+    case ADCS_DOWNLOAD_FILE_LIST_TO_OBC: {
+        status = HAL_ADCS_download_file_list_to_OBC();
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+        set_packet_length(packet, sizeof(int8_t) + 1);
+        break;
+    }
+
+    case ADCS_DOWNLOAD_FILE_TO_OBC: {
+        adcs_file_download_id id;
```

Comment:
Since this id is declared as a stack variable there's no guarantee it makes it to the worker task intact. You will need to malloc it and free it before the task exits or make it static

Reply:
I recall you mentioning this. Is it in the process of passing it into the task that we risk losing it if it's just a stack variable?

Reply:
When it passes to the task the struct is in the correct state, technically. When the creator task gets scheduled again it will start returning from its functions, unwinding its stack along the way. Eventually it gets back to the service_app, returns from that, and the data is no longer available. Sure the created task will have a valid pointer to the memory that was used by the struct at the time, but that memory will get overwritten when the creator calls another function